Modules/_cursesmodule.c

```diff
@@ -1831,62 +1831,71 @@ PyCursesWindow_GetStr(PyObject *op, PyObject *args)
     PyCursesWindowObject *self = _PyCursesWindowObject_CAST(op);
 
     int x, y, n;
-    char rtn[1024]; /* This should be big enough.. I hope */
-    int rtn2;
+    int rtn;
+
+    /* could make the buffer size larger/dynamic */
+    const int max_buf_size = 2048;
+    PyObject *result = PyBytes_FromStringAndSize(NULL, max_buf_size);
+    char *buf = PyBytes_AS_STRING(result);
 
     switch (PyTuple_Size(args)) {
     case 0:
         Py_BEGIN_ALLOW_THREADS
-        rtn2 = wgetnstr(self->win,rtn, 1023);
+        rtn = wgetnstr(self->win, buf, max_buf_size - 1);
         Py_END_ALLOW_THREADS
         break;
     case 1:
-        if (!PyArg_ParseTuple(args,"i;n", &n))
+        if (!PyArg_ParseTuple(args, "i;n", &n))
             return NULL;
         if (n < 0) {
             PyErr_SetString(PyExc_ValueError, "'n' must be nonnegative");
             return NULL;
         }
         Py_BEGIN_ALLOW_THREADS
-        rtn2 = wgetnstr(self->win, rtn, Py_MIN(n, 1023));
+        rtn = wgetnstr(self->win, buf, Py_MIN(n, max_buf_size - 1));
         Py_END_ALLOW_THREADS
         break;
     case 2:
         if (!PyArg_ParseTuple(args,"ii;y,x",&y,&x))
             return NULL;
         Py_BEGIN_ALLOW_THREADS
 #ifdef STRICT_SYSV_CURSES
-        rtn2 = wmove(self->win,y,x)==ERR ? ERR : wgetnstr(self->win, rtn, 1023);
+        rtn = wmove(self->win,y,x)==ERR ? ERR : wgetnstr(self->win, rtn, max_buf_size - 1);
 #else
-        rtn2 = mvwgetnstr(self->win,y,x,rtn, 1023);
+        rtn = mvwgetnstr(self->win,y,x,buf, max_buf_size - 1);
 #endif
         Py_END_ALLOW_THREADS
         break;
     case 3:
-        if (!PyArg_ParseTuple(args,"iii;y,x,n", &y, &x, &n))
+        if (!PyArg_ParseTuple(args, "iii;y,x,n", &y, &x, &n))
```

Comment:
```suggestion
        if (!PyArg_ParseTuple(args,"iii;y,x,n", &y, &x, &n))
```

Let's avoid unrelated changes. This file has a lot of such typos and I don't want to fix them one by one. I'll anyway rewrite this function later with argument clinic so no need for more.